okta_openvpn.py

```diff
@@ -151,14 +152,22 @@ def preauth(self):
         path = "/authn"
         data = {
             'username': self.username,
-            'password': self.password,
+            'password': self.password
         }
+
+        if self.device_token_generator == 'UsernameIpGenerator':
+            data['context'] = {
+                'deviceToken': '%s:%s' % (self.username[0:16], self.client_ipaddr)
+            }
+
         return self.okta_req(path, data)
 
     def doauth(self, fid, state_token):
-        path = "/authn/factors/{fid}/verify".format(fid=fid)
+        rememberDevice = (self.device_token_generator != None)
+        path = "/authn/factors/{fid}/verify?rememberDevice={rememberDevice}".format(
+            fid=fid, rememberDevice=rememberDevice
+        )
         data = {
-            'fid': fid,
```

Comment:
According to the API docs fid is a url parameter and including it in the body is redundant.